### PR TITLE
Fix #775

### LINKED ISSUE #775: The package at "node_modules/parse/lib/react-native/ParseFile.js" attempted to import the Node standard library module "https"
### Issue Description
With changes come with #765 causes this issue. There was a similiar issue caused by `http` package. Please see #768 . It supposed to be fixed but now i have this.

There is no problem with version `2.2.1`.

### Steps to reproduce

Try to use `parse/react-native` with version >= 2.3

### Actual Outcome

> The package at "node_modules/parse/lib/react-native/ParseFile.js" attempted to import the Node standard library module "https". It failed because React Native does not include the Node standard library.

### Environment Setup

- **Server**
  - parse-server version: `2.8.4`
  - Operating System:     [FILL THIS OUT]
  - Hardware:             [FILL THIS OUT]
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): `Back4App`

- **JS SDK**
  - JS SDK version: `2.3.1`
  - Application? (Browser, Node, React-Native, etc): `React Native`

### PR DESCRIPTION
**Pull request by dplewis**
Closes: #775 

Prevent modules from compiling in react-native.

@alioguzhan Can you check if this fix works for you?

### CODE REVIEW

File: src/ParseFile.js [hunk 1]
```diff
@@ -317,25 +317,27 @@ const DefaultController = {
     if (XHR) {
       return this.downloadAjax(uri);
     }
-    if (process.env.PARSE_BUILD === 'browser') {
+    if (process.env.PARSE_BUILD === 'browser' && !XHR) {
```
Comment (acinader) [inline src/ParseFile.js]: you could just change the if to and `else if` and drop the `&& !XHR`

File: src/ParseFile.js [hunk 2]
```diff
@@ -317,25 +317,27 @@ const DefaultController = {
...
-        : require('http');
-      client.get(uri, (resp) => {
-        resp.setEncoding('base64');
-        let base64 = '';
-        resp.on('data', (data) => base64 += data);
-        resp.on('end', () => {
-          resolve({
-            base64,
-            contentType: resp.headers['content-type'],
+    if (process.env.PARSE_BUILD === 'node') {
```
Comment (acinader) [inline src/ParseFile.js]: is it possilbe for this if to be false and neither of the two ifs above to be true?

if so, how to handle?  if not, then we shouldn't need this test?

Comment (dplewis) [reply src/ParseFile.js]: The first if handles react-native and browser
The second for unsupported browsers
The third is for node only. (Note babel trick here. This won't end up in the react-native build)

You can check by forcing this branch. `npm run build` and check `/lib/xxx/ParseFile.js` and notice the difference

**Comment by alioguzhan** [comment]
Thanks for the quick fix! I will check it today and will let you know. @dplewis

**Comment by alioguzhan** [comment]
Sorry @dplewis , I couldn't have the chance to test this fix. But looks fine to me.  👍

**Pull request merged**